calculator.cpp

```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+using namespace std;
+
+template <typename T> 
+T sum(T& , T& );
+
+template <typename T>
+T mult(T& , T& );
+
+template <typename T>
+T diff(T& , T&);
+
+template <typename T>
+T div(T& , T&);
+
+
+template <typename T>
+T input(){
+    int v;
+    cout << "introduct a value from keyboard: ";
+    cin >> v;
+    cout << endl;
+    return v;
+}
+
+
+int main(){
+    
+    auto x = input<int>();
+    auto y = input<int>();
```

Comment:
`auto`, very very nice 🚀 

in realtà però ti consiglierei di passare la variabile x per riferimento alla funzione input() in modo tale da evitare l'assegnazione

però usare "auto" è una best practice ti C/C++, ottimo 🚀

Reply:
Grazie mille del consiglio, in effetti hai ragione non avevo pensato a passare la varaibile x per riferimento.